extensions/amp-byside-content/0.1/utils.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * Returns a function, that, as long as it continues to be invoked, will not
+ * be triggered. The function will be called after it stops being called for
+ * N milliseconds. If `immediate` is passed, trigger the function on the
+ * leading edge, instead of the trailing.
+ * @param {Function} func
+ * @param {number} wait
+ * @param {boolean=} immediate
+ */
+export function debounce(func, wait, immediate) {
```

Comment:
We already have a debounce in [src/utils/rate-limit.js](https://github.com/ampproject/amphtml/blob/7ca49a954f10d3ed531f015c144161410f950921/src/utils/rate-limit.js#L56-L93).